### PR TITLE
Add ReadDir and fix test directory deletion

### PR DESCRIPTION
**Pull request by jfontan**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@ before_script:
   - make container
 
 script:
+  - cp Gopkg.testing Gopkg.toml
+  - make godep
```
Comment (kuba--) [inline .travis.yml]: yeah , we need `go mod`

File: filesystem.go [hunk 2]
```diff
@@ -135,10 +134,29 @@ func (g *FS) Join(elem ...string) string {
...
+	defer d.Close()
+
+	files, err := d.Readdir(0)
+	if err != nil {
+		return nil, err
+	}
+
+	// gluster Readdir returns also "." and ".."
+	clean := make([]os.FileInfo, 0, len(files))
+	for _, f := range files {
```
Comment (kuba--) [inline filesystem.go]: maybe instead of copy objects to the new slice, we can find indexes with "." and/or ".." and at the end return the reference to the shifted slice 
for instance:
```go
        n := len(files)
	for i := 0; i < n; i++ {
		if files[i].Name() != "." && files[i].Name() != ".." {
                       // swap with the last element
			files[i], files[n-1] = files[n-1], files[i]
			n--
			i--
		}
	}
       return files[0:n]
```
WDYT? Maybe it sounds like a micro optimization but if you have many files, it can avoid lot of reallocations.

Comment (jfontan) [reply filesystem.go]: Added the optimization and an extra check to test empty directories. Also fixed a bug deleting directories.
